src/storage/jni.rs

```diff
@@ -0,0 +1,320 @@
+            .into_inner()
+    } else {
+        null_mut()
+    }
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Storage_nativeCreateTransaction(
+    _env: JNIEnv,
+    _storage: JObject
+) -> jobject
+{
+    let storage = unwrap_ptr::<Storage>(&_env, _storage);
+
+    if let Ok(transaction) = storage.create_transaction(){
+        let transaction_class = _env.find_class("com/horizen/storage/Transaction")
+            .expect("Should be able to find class Transaction");
+        create_java_object(&_env, &transaction_class, transaction)
+    } else {
+        null_mut()
+    }
+}
+// ------------------------------------- Transaction JNI wrappers -------------------------------------
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Transaction_nativeClose(
+    _env: JNIEnv,
+    _class: JClass,
+    _transaction: *mut Transaction,
+)
+{
+    if !_transaction.is_null(){
+        drop(unsafe { Box::from_raw(_transaction) })
+    }
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Transaction_nativeGet(
+    _env: JNIEnv,
+    _transaction: JObject,
+    _cf: JObject,
+    _key: jbyteArray
+) -> jbyteArray
+{
+    let transaction = unwrap_ptr::<Transaction>(&_env, _transaction);
+    let cf = unwrap_ptr::<ColumnFamily>(&_env, _cf);
+
+    let key = _env.convert_byte_array(_key)
+        .expect("Should be able to convert to Rust byte array");
+
+    if let Some(value) = transaction.get_cf(cf, key.as_slice()){
+        _env.byte_array_from_slice(value.as_slice())
+            .expect("Should be able to convert Rust slice into jbytearray")
+    } else {
+        null_mut()
+    }
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Transaction_nativeMultiGet(
+    _env: JNIEnv,
+    _transaction: JObject,
+    _cf: JObject,
+    _keys: jobjectArray
+) -> jobject
+{
+    let transaction = unwrap_ptr::<Transaction>(&_env, _transaction);
+    let cf = unwrap_ptr::<ColumnFamily>(&_env, _cf);
+    let keys = java_array_to_vec_byte(&_env, _keys);
+
+    let key_values = transaction.multi_get_cf(
+        cf,
+        keys.iter().map(|k|k.as_slice()).collect::<Vec<_>>().as_slice()
+    );
+    map_to_java_map(&_env, &key_values)
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Transaction_nativeIsEmpty(
+    _env: JNIEnv,
+    _transaction: JObject,
+    _cf: JObject,
+) -> jboolean {
+    let transaction = unwrap_ptr::<Transaction>(&_env, _transaction);
+    let cf = unwrap_ptr::<ColumnFamily>(&_env, _cf);
+
+    if let Ok(is_empty) = transaction.is_empty_cf(cf){
+        is_empty as jboolean
+    } else {
+        JNI_TRUE
+    }
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Transaction_nativeCommit(
+    _env: JNIEnv,
+    _transaction: JObject,
+)-> jboolean {
+    let transaction = unwrap_ptr::<Transaction>(&_env, _transaction);
+    if let Ok(()) = transaction.commit(){
+        JNI_TRUE
+    } else {
+        JNI_FALSE
```

Comment:
What kind of Error `transaction.commit()` may return to us?
Similar to what was mentioned in other places, I think, it is better for us to throw an exception in case of any error, so user outside will be able to know the reason as well. With `boolean` result type it may be hard to debug what is the real cause of failure.